packages/components/src/internal/components/chart/ChartSettingsPanel.tsx

```diff
@@ -0,0 +1,411 @@
+                setSizes(current => ({ ...current, width: '' }));
+            }
+
+            setUseFullWidth(checked);
+        },
+        [setChartConfig]
+    );
+    const onNumberChange = useCallback((name, value) => setSizes(current => ({ ...current, [name]: value })), []);
+    const onBlur = useCallback(() => {
+        setChartConfig(current => {
+            const [height, heightChanged] = changedValue(sizes.height, current.height);
+            const [width, widthChanged] = changedValue(sizes.width, current.width);
+
+            if (!heightChanged && !widthChanged) return current;
+
+            return {
+                ...current,
+                height: heightChanged ? height : current.height,
+                width: widthChanged ? width : current.width,
+            };
+        });
+    }, [setChartConfig, sizes]);
+
+    return (
+        <>
+            {/* intentionally not setting form-group class on the div below */}
+            <div className="row">
+                <div className="col-xs-6">
+                    <NumberInput
+                        disabled={useFullWidth}
+                        label="Width (px)"
+                        name="width"
+                        onBlur={onBlur}
+                        onChange={onNumberChange}
+                        value={sizes.width}
+                    />
+                </div>
+                <div className="col-xs-6">
+                    <NumberInput
+                        disabled={false}
+                        label="Height (px)"
+                        name="height"
+                        onBlur={onBlur}
+                        onChange={onNumberChange}
+                        value={sizes.height}
+                    />
+                </div>
+            </div>
+            <div className="form-group row">
+                <div className="col-xs-12">
+                    <CheckboxLK checked={useFullWidth} name="use-full-width" onChange={onCheckboxChange}>
+                        Full Width
+                    </CheckboxLK>
+                </div>
+            </div>
+        </>
+    );
+});
+SizeInputs.displayName = 'SizeInputs';
+
+interface ChartTypeOptionRendererProps {
+    chartType: ChartTypeInfo;
+    isValueRenderer: boolean;
+}
+const ChartTypeOptionRenderer: FC<ChartTypeOptionRendererProps> = memo(({ chartType, isValueRenderer }) => {
+    const icon = ICONS[chartType.name];
+    const isSvg = icon.endsWith('.svg');
+    const className = classNames('chart-builder-type-option', { 'chart-builder-type-option--value': isValueRenderer });
+    return (
+        <span className={className} data-chart-type={chartType.name}>
+            {isSvg && (
+                <SVGIcon height={null} iconDir="visualization/report" iconSrc={icon.replace('.svg', '')} width={16} />
+            )}
+            {!isSvg && <span className={`fa ${icon}`} />}
+            <span>{chartType.title}</span>
```

Comment:
nice! glad we were able to include the icons in the input options, it looks great.